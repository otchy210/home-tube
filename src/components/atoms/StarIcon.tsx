import React from 'react';
import SelectedStar from '../../images/star-selected.svg';
import UnselectedStar from '../../images/star-unselected.svg';
import VoidStar from '../../images/star-void.svg';

export type StarIconVariant = 'selected' | 'unselected' | 'void';

type Props = {
    variant: StarIconVariant;
    size: number;
};

const StarIcon: React.FC<Props> = ({ variant, size }: Props) => {
    const sizes = { width: size, height: size };
    switch (variant) {
        case 'selected':
            return <SelectedStar {...sizes} />;
        case 'unselected':
            return <UnselectedStar {...sizes} />;
        case 'void':
            return <VoidStar {...sizes} />;
    }
};

export default StarIcon;
