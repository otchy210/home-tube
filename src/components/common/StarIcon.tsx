import React from 'react';
import styled from 'styled-components';
import { StarSelectedSvg } from '../images/StarSelectedSvg';
import { StarUnselectedSvg } from '../images/StarUnselectedSvg';
import { StarVoidSvg } from '../images/StarVoidSvg';

const IconWrapper = styled.div`
    display: inline-box;
    pointer-events: none;
`;

export type StarIconVariant = 'selected' | 'unselected' | 'void';

type Props = {
    variant: StarIconVariant;
    size: number;
};

const StarIcon: React.FC<Props> = ({ variant, size }: Props) => {
    const sizes = { width: size, height: size };
    switch (variant) {
        case 'selected':
            return (
                <IconWrapper>
                    <StarSelectedSvg {...sizes} />
                </IconWrapper>
            );
        case 'unselected':
            return (
                <IconWrapper>
                    <StarUnselectedSvg {...sizes} />
                </IconWrapper>
            );
        case 'void':
            return (
                <IconWrapper>
                    <StarVoidSvg {...sizes} />
                </IconWrapper>
            );
    }
};

export default StarIcon;
