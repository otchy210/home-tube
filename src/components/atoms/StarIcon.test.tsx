import { render } from '@testing-library/react';
import React from 'react';
import StarIcon, { StarIconVariant } from './StarIcon';

describe('StarIcon', () => {
    const variantOptions = ['selected', 'unselected', 'void'] as StarIconVariant[];
    it.each(variantOptions)('renders properly with variant %s', (variant) => {
        const { container } = render(<StarIcon variant={variant} size={16} />);
        expect(container).toMatchSnapshot();
    });
});
