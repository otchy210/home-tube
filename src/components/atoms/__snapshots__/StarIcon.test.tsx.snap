// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarIcon renders properly with variant selected 1`] = `
<div>
  <svgr-mock
    height="16"
    width="16"
  />
</div>
`;

exports[`StarIcon renders properly with variant unselected 1`] = `
<div>
  <svgr-mock
    height="16"
    width="16"
  />
</div>
`;

exports[`StarIcon renders properly with variant void 1`] = `
<div>
  <svgr-mock
    height="16"
    width="16"
  />
</div>
`;
